diff --git a/drivers/net/wireless/ath/wcn36xx/main.c b/drivers/net/wireless/ath/wcn36xx/main.c
index d4306fa..8cc96d4 100644
--- a/drivers/net/wireless/ath/wcn36xx/main.c
+++ b/drivers/net/wireless/ath/wcn36xx/main.c
@@ -135,13 +135,10 @@ static struct ieee80211_supported_band wcn_band_2ghz = {
 	.n_bitrates	= ARRAY_SIZE(wcn_2ghz_rates),
 	.ht_cap		= {
 		.cap =	IEEE80211_HT_CAP_GRN_FLD |
-			IEEE80211_HT_CAP_SGI_20 |
 			IEEE80211_HT_CAP_DSSSCCK40 |
-			IEEE80211_HT_CAP_LSIG_TXOP_PROT |
-			IEEE80211_HT_CAP_SGI_40 |
-			IEEE80211_HT_CAP_SUP_WIDTH_20_40,
+			IEEE80211_HT_CAP_LSIG_TXOP_PROT,
 		.ht_supported = true,
-		.ampdu_factor = IEEE80211_HT_MAX_AMPDU_64K,
+		.ampdu_factor = IEEE80211_HT_MAX_AMPDU_32K,
 		.ampdu_density = IEEE80211_HT_MPDU_DENSITY_16,
 		.mcs = {
 			.rx_mask = { 0xff, 0, 0, 0, 0, 0, 0, 0, 0, 0, },
@@ -309,6 +306,8 @@ static void wcn36xx_change_ps(struct wcn36xx *wcn, bool enable)
 	struct ieee80211_vif *vif = NULL;
 	struct wcn36xx_vif *tmp;
 
+	return;
+
 	list_for_each_entry(tmp, &wcn->vif_list, list) {
 		vif = wcn36xx_priv_to_vif(tmp);
 		if (enable && !wcn->sw_scan) {
@@ -414,9 +413,7 @@ static int wcn36xx_config(struct ieee80211_hw *hw, int radio_idx, u32 changed)
 		wcn36xx_change_ps(wcn, hw->conf.flags & IEEE80211_CONF_PS);
 
 	if (changed & IEEE80211_CONF_CHANGE_IDLE) {
-		if (hw->conf.flags & IEEE80211_CONF_IDLE)
-			wcn36xx_smd_enter_imps(wcn);
-		else
+		if (!(hw->conf.flags & IEEE80211_CONF_IDLE))
 			wcn36xx_smd_exit_imps(wcn);
 	}
 
@@ -896,8 +893,11 @@ static void wcn36xx_bss_info_changed(struct ieee80211_hw *hw,
 			 * place where AID is available.
 			 */
 			wcn36xx_smd_config_sta(wcn, vif, sta);
+			/* Disable beacon filter for stability */
+			/*
 			if (vif->type == NL80211_IFTYPE_STATION)
 				wcn36xx_smd_add_beacon_filter(wcn, vif);
+			*/
 			wcn36xx_enable_keep_alive_null_packet(wcn, vif);
 		} else {
 			wcn36xx_dbg(WCN36XX_DBG_MAC,
@@ -1106,9 +1106,6 @@ static int wcn36xx_suspend(struct ieee80211_hw *hw, struct cfg80211_wowlan *wow)
 		if (ret)
 			goto out;
 		ret = wcn36xx_smd_ipv6_ns_offload(wcn, vif, true);
-		if (ret)
-			goto out;
-		ret = wcn36xx_smd_gtk_offload(wcn, vif, true);
 		if (ret)
 			goto out;
 		ret = wcn36xx_smd_set_power_params(wcn, true);
@@ -1140,8 +1137,6 @@ static int wcn36xx_resume(struct ieee80211_hw *hw)
 	if (vif) {
 		wcn36xx_smd_host_resume(wcn);
 		wcn36xx_smd_set_power_params(wcn, false);
-		wcn36xx_smd_gtk_offload_get_info(wcn, vif);
-		wcn36xx_smd_gtk_offload(wcn, vif, false);
 		wcn36xx_smd_ipv6_ns_offload(wcn, vif, false);
 		wcn36xx_smd_arp_offload(wcn, vif, false);
 	}

diff --git a/drivers/net/wireless/ath/wcn36xx/smd.c b/drivers/net/wireless/ath/wcn36xx/smd.c
index ff10c63..2e15a34 100644
--- a/drivers/net/wireless/ath/wcn36xx/smd.c
+++ b/drivers/net/wireless/ath/wcn36xx/smd.c
@@ -3323,8 +3323,13 @@ int wcn36xx_smd_rsp_process(struct rpmsg_device *rpdev,
 		wcn36xx_dbg(WCN36XX_DBG_HAL, "indication arrived\n");
 		break;
 	default:
-		wcn36xx_err("SMD_EVENT (%d) not supported\n",
-			      msg_header->msg_type);
+		if (msg_header->msg_type == 312) {
+        	wcn36xx_dbg(WCN36XX_DBG_HAL, "SMD_EVENT (312) not supported (ignored)\n");
+    	} else {
+        	wcn36xx_err("SMD_EVENT (%d) not supported\n", 
+					msg_header->msg_type);
+    	}
+    	break;
 	}
 
 	return 0;
